Applications/FileIO/SWMM/SWMMInterface.cpp

```diff
@@ -0,0 +1,1248 @@
+                return false;
+        }
+        /* todo: check if needed
+        if (line == "[VERTICES]")
+        {
+            INFO ("Reading vertices...");
+            if (!readCoordinates(in, nodes, _id_nodename_map))
+                return false;
+        }
+        */
+        if (line == "[SYMBOLS]")
+        {
+            INFO ("Reading symbols...");
+            std::vector<GeoLib::Point*> points;
+            std::vector<std::string> names;
+            if (!readCoordinates(in,points, names))
+                return false;
+            for (std::size_t i=0; i<points.size(); ++i)
+            {
+                GeoLib::Station stn (points[i], names[i]);
+                _rain_gauges.push_back(std::pair<GeoLib::Station, std::string>(stn, ""));
+            }
+        }
+    }
+
+    if (nodes.empty())
+        return false;
+
+    // After end of file is reached, create name-id-map and
+    // start reading again to get line elements and node data.
+    std::map< std::string, std::size_t> name_id_map;
+    std::size_t const n_nodes (nodes.size());
+    for (std::size_t i=0; i<n_nodes; ++i)
+        name_id_map[_id_nodename_map[i]] = i;
+    in.clear();
+    in.seekg(0, in.beg);
+
+    std::vector< MeshLib::Element* > elements;
+    std::vector<double> max_depth;
+    max_depth.resize(n_nodes);
+    std::size_t const n_types = 3;
+    std::array< std::size_t, n_types> n_elem_types;
+    while ( getline(in, line) )
+    {
+        if (line == "[RAINGAGES]")
+        {
+            if (!_rain_gauges.empty())
+                addRainGaugeTimeSeriesLocations(in);
+        }
+        if (line == "[SUBCATCHMENTS]")
+        {
+            INFO ("Reading subcatchment information...");
+            if (!readSubcatchments(in, name_id_map))
+                return false;
+        }
+        if (line == "[SUBAREAS]")
+        {
+            // more subcatchment variables, not yet implemented
+        }
+        if (line == "[INFILTRATION]")
+        {
+            // more subcatchment variables, not yet implemented
+        }
+        if (line == "[JUNCTIONS]")
+        {
+            INFO ("Reading junctions...");
+            if (!readNodeData(in, nodes, name_id_map, max_depth, true))
+                return false;
+        }
+        else if (line == "[OUTFALLS]")
+        {
+            INFO ("Reading outfalls...");
+            if (!readNodeData(in, nodes, name_id_map, max_depth, false))
+                return false;
+        }
+        else if (line == "[STORAGE]")
+        {
+            INFO ("Reading storages...");
+            if (!readNodeData(in, nodes, name_id_map, max_depth, true))
+                return false;
+        }
+        else if (line == "[CONDUITS]")
+        {
+            INFO ("Reading conduits...");
+            if (!readLineElements(in, elements, nodes, name_id_map))
+                return false;
+            n_elem_types[0] = elements.size();
```

Comment:
It's just used in this one place in this one method to distinguish between different input sources. n_elem_types is a std::array so case of index errors you'd get compile errors.